Sources/TensorFlow/Layers/Recurrent.swift

```diff
@@ -703,3 +687,54 @@ public typealias SimpleRNNCell = BasicRNNCell
 
 @available(*, deprecated, renamed: "BasicRNN")
 public typealias SimpleRNN = BasicRNN
+
+// - MARK: Workaround helpers.
+
+fileprivate extension Array where Element: Differentiable {
+  /// Returns a reversed copy of `self`.
+  ///
+  /// This has a custom derivative, which works around the SR-13945 segfault that you would
+  /// encounter if you tried to implement this at the callsite using a for loop.
+  @differentiable
+  func differentiableReversed() -> Self {
+    .init(self.reversed())
+  }
+
+  @derivative(of: differentiableReversed)
+  func vjpDifferentiableReversed()
+    -> (value: Self, pullback: (TangentVector) -> TangentVector)
+  {
+    return (self.differentiableReversed(), { .init(.init($0.base.reversed())) })
+  }
```

Comment:
No urgency: I wonder if this derivative could be defined as `@derivative(of: reversed)` and put in the `_Differentiation` module?

Reply:
That would be nice. `reversed()` returns a `ReversedCollection` so we would need to make that differentiable too.

Reply:
reverse() is mutating. Perhaps that could be used instead?